lifemonitor/api.yaml

```diff
@@ -183,10 +431,150 @@ components:
         - version
         - roc_link
 
+    WorkflowStatus:
+      type: object
+      properties:
+        workflow:
+          schema:
+            type: object
+            properties:
+              uuid:
+                type: string
+              version:
+                type: string
+              name:
+                type: string
+            required:
+              - uuid
+              - version
+        aggregate_test_status:
+          $ref: '#/components/schemas/AggregateTestStatus'
+        latest_builds:
+          description: "Latest builds, one for each test instance"
+          type: array
+          items:
+            $ref: '#/components/schemas/BuildSummary'
+
+    AggregateTestStatus:
+      type: string
+      enum:
+        - "all-passing"
+        - "some-passing"
+        - "all-failing"
+        - "not-available"
+
     TestSuite:
       type: object
       properties:
+        uuid:
+          type: string
         test_suite_metadata:
           type: object
+          description: "To be determined."
```

Comment:
There is currently a mismatch between https://github.com/crs4/life_monitor/wiki/Test-Metadata-Draft-Spec and how we are structuring the API.

In the specs we assume that "a Workflow RO-Crate containing test information can always be obtained from the WFHub", under the hypothesis that the WorkflowHub will add a way for its users to provide `test-metadata.json` and any additional files in cases where the workflow is not uploaded as an RO-Crate from the beginning. This assumption allows to cover all cases, including the "internal testing" where the additional user-provided test files and directories will be in the RO-Crate.

In the API, on the other hand (even before this PR, actually), we are providing a POST call to directly add test suites via the LM. However, for now we are focusing on tests being executed on an unmanaged instance, where only the metadata is needed. Do we also want to allow adding test suites in the case of user-defined tests? If so, we'd need a way to support adding all the additional test files and directories and make sure that the `payload` element links to them in a consistent way. It seems a bit like reinventing the RO-Crate.

Also, `TestSuite` has a `test_suite_metadata` field, but part of the metadata (the `instance` part) is currently in the `SuiteInstance` schema. Could this lead to some kind of duplication?

Reply:
I don't think there's really a mismatch with the Workflow-RO-create test metadata, but I agree that -- at least at this stage -- supporting adding tests directly through LM would introduce significant complexity.  I just removed it.

Regarding your point about potential duplication between `TestSuite` and `SuiteInstance`, that's intentional.  I proposed it this way to reduce the number of API requests required for presumably typical usage -- i.e., one retrieves a TestSuite and receives it complete of the SuiteInstances where it runs, instead of having to retrieve them in subsequent calls.